6-AdvancedScenarios/1-call-api-obo/README.md

```diff
@@ -1,427 +1,361 @@
+    1. Select **Set** next to the **Application ID URI** to generate a URI that is unique for this app.
+    1. For this sample, accept the proposed Application ID URI (`api://{clientId}`) by selecting **Save**.
+        > :information_source: Read more about Application ID URI at [Validation differences by supported account types (signInAudience)](https://docs.microsoft.com/azure/active-directory/develop/supported-accounts-validation).
+
+##### Publish Delegated Permissions
+
+1. All APIs must publish a minimum of one [scope](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code), also called [Delegated Permission](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#permission-types), for the client apps to obtain an access token for a *user* successfully. To publish a scope, follow these steps:
+1. Select **Add a scope** button open the **Add a scope** screen and Enter the values as indicated below:
+    1. For **Scope name**, use `access_graph_on_behalf_of_user`.
+    1. Select **Admins and users** options for **Who can consent?**.
+    1. For **Admin consent display name** type in *Access Microsoft Graph as the signed-in user*.
+    1. For **Admin consent description** type in *Allow the app to access Microsoft Graph Api as the signed-in user*.
+    1. For **User consent display name** type in *Access Microsoft Graph on your behalf*.
+    1. For **User consent description** type in *Allow the Microsoft Graph APi on your behalf.*.
+    1. Keep **State** as **Enabled**.
+    1. Select the **Add scope** button on the bottom to save this scope.
+1. Select the **Manifest** blade on the left.
+    1. Set `accessTokenAcceptedVersion` property to **2**.
+    1. Select on **Save**.
+
+> :information_source:  Follow [the principle of least privilege when publishing permissions](https://learn.microsoft.com/security/zero-trust/develop/protected-api-example) for a web API.
+
+##### Configure Optional Claims
+
+1. Still on the same app registration, select the **Token configuration** blade to the left.
+1. Select **Add optional claim**:
+   1. Select **optional claim type**, then choose **Access**.
+      1. Select the optional claim **idtyp**.
+      > Indicates token type. This claim is the most accurate way for an API to determine if a token is an app token or an app+user token. This is not issued in tokens issued to users.
+   1. Select **Add** to save your changes.
+1. Still on the same app registration, select the **Manifest** blade to the left.
+   1. Set the **optionalClaims** property as shown below to request client capabilities claim *xms_cc*:
+
+   ```json
+      "optionalClaims": 
+      {
+      "accessToken": [
+         {
+            "additionalProperties": [],
+            "essential": false,
+            "name": "xms_cc",
+            "source": null
+         }
+      ],
+      "idToken": [],
+      "saml2Token": []
+      }
+    ```
```

Comment:
please open a task on code gen

Create a method named GenerateOptionalClaimManualSteps(string claimName, string tokenType) 
then
use a switch case in CodeGeneratorHelper.GetOptionalClaimDescription() for claims such as xms_cc to call this new method that prints these steps in readme

Reply:
Opened an issue in [codegen](https://github.com/Azure-Samples/GenerateScriptsForSample/issues/79)